File: 1.0/src/cz/mg/vulkan/PFNvkBindBufferMemory.java
package cz.mg.vulkan;

public class PFNvkBindBufferMemory extends VkFunctionPointer {
    public PFNvkBindBufferMemory() {
    }

    protected PFNvkBindBufferMemory(VkMemory vkmemory) {
        super(vkmemory);
    }

    protected PFNvkBindBufferMemory(VkMemory vkmemory, long vkaddress) {
        super(vkmemory, vkaddress);
    }



    public PFNvkBindBufferMemory(long value) {
        setValue(value);
    }

    public PFNvkBindBufferMemory(VkInstance instance) {
        super(instance, new VkString("vkBindBufferMemory"));
    }

    public void call(VkDevice device, VkBuffer buffer, VkDeviceMemory memory, VkDeviceSize memoryOffset, VkResult rval){
        callNative(getValue(), device != null ? device.getVkAddress() : VkPointer.getNullAddressNative(), buffer != null ? buffer.getVkAddress() : VkPointer.getNullAddressNative(), memory != null ? memory.getVkAddress() : VkPointer.getNullAddressNative(), memoryOffset != null ? memoryOffset.getVkAddress() : VkPointer.getNullAddressNative(), rval != null ? rval.getVkAddress() : VkPointer.getSinkAddressNative());
    }

    protected static native void callNative(long vkaddress, long device, long buffer, long memory, long memoryOffset, long rval);


    public int call(VkDevice device, VkBuffer buffer, VkDeviceMemory memory, long memoryOffset){
        return callSimplifiedNative(getValue(), device != null ? device.getVkAddress() : VkPointer.getNullAddressNative(), buffer != null ? buffer.getVkAddress() : VkPointer.getNullAddressNative(), memory != null ? memory.getVkAddress() : VkPointer.getNullAddressNative(), memoryOffset);
    }

    protected static native int callSimplifiedNative(long vkaddress, long device, long buffer, long memory, long memoryOffset);

}

File: 1.0/src/c/cz/mg/vulkan/PFN_vkBindBufferMemory.c
#include <jni.h>
#include <stdlib.h>
#include <string.h>
#include <vulkan/vulkan.h>

jlong jniPointerToLong(const void* p);
void* jniLongToPointer(jlong l);
jlong jniFunctionPointerToLong(PFN_vkVoidFunction p);
PFN_vkVoidFunction jniLongToFunctionPointer(jlong l);
void jniThrowException(JNIEnv* env, const char* message);

void Java_cz_mg_vulkan_PFNvkBindBufferMemory_callNative(JNIEnv* env, jclass jc, jlong address, jlong device, jlong buffer, jlong memory, jlong memoryOffset, jlong rval){
    (void)env;
    (void)jc;
    PFN_vkBindBufferMemory f = (PFN_vkBindBufferMemory)jniLongToFunctionPointer(address);
    VkResult* rvalAddress = jniLongToPointer(rval);
    *rvalAddress = f(
        *((VkDevice*)jniLongToPointer(device)),
        *((VkBuffer*)jniLongToPointer(buffer)),
        *((VkDeviceMemory*)jniLongToPointer(memory)),
        *((VkDeviceSize*)jniLongToPointer(memoryOffset))
    );
}


jint Java_cz_mg_vulkan_PFNvkBindBufferMemory_callSimplifiedNative(JNIEnv* env, jclass jc, jlong address, jlong device, jlong buffer, jlong memory, jlong memoryOffset){
    (void)env;
    (void)jc;
    PFN_vkBindBufferMemory f = (PFN_vkBindBufferMemory)jniLongToFunctionPointer(address);
    return f(
        *((VkDevice*)jniLongToPointer(device)),
        *((VkBuffer*)jniLongToPointer(buffer)),
        *((VkDeviceMemory*)jniLongToPointer(memory)),
        (VkDeviceSize)memoryOffset
    );
}

